### PR TITLE
Add support for setup task

### LINKED ISSUE #111: Introduce mechanism to configure deployed resources.
**Goal**
Introduce new `Setup` annotation with one parameter that defines a setup class. The purpose of the class is to configure resources or application running on them before tests are executed.

**Requirements**
- Setup class provide methods for set-up and tear-down
- Setup class perform set up operation once all resources are deployed
- Set up operation is executed right before WAR/JAR/EAR are deployed (once Sunstone supports that)
- Tear down operation is executed once all tests are finished
- An user can inject both static and non-static resources into setup class
- Constructor with zero parameters is required for setup class

### PR DESCRIPTION
**Pull request by istraka**
Fixes #111 

FYI @mchoma

### CODE REVIEW

File: clouds/clouds-core/src/main/java/sunstone/core/SunstoneExtension.java [hunk 1]
```diff
@@ -69,10 +72,44 @@ public void beforeAll(ExtensionContext ctx) throws Exception {
...
+            handleSetup(ctx);
+        }
         performDeploymentOperation(ctx);
         injectStaticResources(ctx, ctx.getRequiredTestClass());
     }
 
+    static void handleSetup(ExtensionContext ctx) throws IllegalArgumentSunstoneException {
+        SunstoneStore store = new SunstoneStore(ctx);
+        if (ctx.getRequiredTestClass().getAnnotationsByType(Setup.class).length != 1) {
+            throw new RuntimeException();
```
Comment (mnovak1) [inline clouds/clouds-core/src/main/java/sunstone/core/SunstoneExtension.java:85]: I think some error message should be present.

File: clouds/clouds-core/src/test/java/sunstone/core/setup/SetupSuite.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package sunstone.core.setup;
+
+import org.junit.platform.suite.api.SelectClasses;
+import org.junit.platform.suite.api.Suite;
+import sunstone.core.setup.suitetests.SetupCleanupSecondTest;
+import sunstone.core.setup.suitetests.SetupFirstTest;
+
+@Suite
+@SelectClasses({SetupFirstTest.class, SetupCleanupSecondTest.class})
```
Comment (mnovak1) [inline clouds/clouds-core/src/test/java/sunstone/core/setup/SetupSuite.java:9]: I'm not sure whether ordering is guarunteed by JUnit 5.

Comment (istraka) [reply clouds/clouds-core/src/test/java/sunstone/core/setup/SetupSuite.java:9]: It is not stated anywhere, but it is an array, hence I assume why the order is the same. But I agree this is not ideal. Here is the issue. https://github.com/wildfly-extras/sunstone/issues/126

File: clouds/clouds-core/src/main/java/sunstone/core/SunstoneExtension.java (hunk 1 above)
Comment (istraka) [reply clouds/clouds-core/src/main/java/sunstone/core/SunstoneExtension.java]: Fixed.

**Review by mnovak1 — APPROVED** [review]
LGTM.

**Comment by mchoma** [comment]
This is OK in my perspective.

**Pull request merged**